app/scripts/modules/core/src/cluster/filter/FilterSearch.tsx

```diff
@@ -5,27 +5,26 @@ import './FilterSearch.less';
 
 export interface IFilterSearchProps {
   helpKey?: string;
-  onBlur: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onBlur: (event: React.FocusEvent<HTMLInputElement>) => void;
   onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
   value: string;
 }
 
-export const FilterSearch = ({ helpKey, onBlur, onSearchChange, value }: IFilterSearchProps) => {
-  return (
-    <div className="filter-search">
-      <form className="horizontal middle space-between" role="form">
-        <div className="form-group nav-search">
-          <input
-            type="search"
-            className="form-control input-med sp-form"
-            value={value}
-            onBlur={onBlur}
-            onChange={onSearchChange}
-            placeholder="Search"
-          />
-        </div>
-        {helpKey && <HelpField id={helpKey} placement="right" />}
-      </form>
-    </div>
-  );
-};
+export const FilterSearch = ({ helpKey, onBlur, onSearchChange, value }: IFilterSearchProps) => (
+  <div className="filter-search">
+    <form className="horizontal middle space-between" role="form">
+      <div className="form-group nav-search">
+        <input
+          type="search"
+          className="form-control input-med sp-form"
+          value={value}
+          onBlur={onBlur}
+          onChange={onSearchChange}
+          placeholder="Search"
+          ng-model={value}
```

Comment:
This is a surprising addition.  Can you explain how this works?  AngularJS shouldn't be compiling this react template, as far as I understand.

Reply:
TL;DR This is not needed if I migrate this whole section to React :)

When an Angular template is a parent of this React component, the controller is set up in a way that needs this in order to update the reference model. The `updateClusterGroups` function in the controller has to run after any filter or search is updated. This uses the controller's context of the `FilterSearchModel` which will use the old filter value. `ng-model` will force the context of the `FilterSearch` to reference this value at all times.